lib/atlas/node.rb

```diff
@@ -18,8 +18,9 @@ class ArityLimitError < ::StandardError; end
 
     attr_reader :children
 
-    def initialize(children = [])
-      @children = NodeChildren.new(children)
```

Comment:
The children nodes instantiated in this manner were not having their parent values set, which was normally done [here](https://github.com/bespokepost/atlas/pull/4/files#diff-eb5d7f38586d4012a70d61a11049e6bfL33).